integration_tests/add-custom-barbell-exercise.integration.test.tsx

```diff
@@ -108,3 +116,107 @@ describe("User Journey: Add Custom Barbell Exercises", () => {
     // but I can't get _anything_ to work, try this again sometime.
   });
 });
+
+describe("User Journey: Can use components to edit from the default values", () => {
+  it("should allow a logged in user to modify every field in the UI", async () => {
+    // Initial render of the page.
+    let page = await EquipmentExercisePage(pageProps);
+    await act(async () => render(page));
+
+    // Find and click the "Add Exercise" button. This will add a db form draft.
+    await act(async () => {
+      const { data: initialDeadlifts } = await supabase
+        .from("exercises")
+        .select("*")
+        .eq(
+          "user_id",
+          USER_ID["add-custom-barbell-exercise.integration.test.tsx"],
+        )
+        .eq("exercise_type", "barbell_deadlift");
+      expect(initialDeadlifts?.length).toBe(0);
+      const addExerciseButton = await waitFor(() =>
+        screen.getByTestId(TestIds.AddExerciseButton),
+      );
+      addExerciseButton.click();
+
+      // Wait for a form draft to exist before re-rendering.
+      await waitFor(async () => {
+        const { data: drafts } = await supabase
+          .from("form_drafts")
+          .select("*")
+          .eq(
+            "user_id",
+            USER_ID["add-custom-barbell-exercise.integration.test.tsx"],
+          );
+        expect(drafts?.length).toBeGreaterThan(0);
+      });
+      page = await EquipmentExercisePage(pageProps);
+      await act(async () => render(page));
+    });
+
+    // Find the select reps component and click the AMRAP toggle
+    await act(async () => {
+      (
+        await Promise.all([
+          waitFor(() => screen.getByTestId(TestIds.SelectRepsAMRAPToggle)),
+          waitFor(() => screen.getByTestId(TestIds.SelectRepsAdd)),
+          waitFor(() => screen.getByTestId(TestIds.ActivePlate(45))),
+          waitFor(() => screen.getByTestId(TestIds.ActivePlate(25))),
+          waitFor(() => screen.getByTestId(TestIds.ActivePlate(10))),
+          waitFor(() => screen.getByTestId(TestIds.ActivePlate(5))),
+          waitFor(() => screen.getByTestId(TestIds.ActivePlate(2.5))),
+          waitFor(() => screen.getByTestId(TestIds.PerceivedEffort("hard"))),
+          waitFor(() => screen.getByTestId(TestIds.WarmupToggle)),
+        ])
+      ).map((e) => e.click());
```

Comment:
This is arguably too clever, but it's just finding all of those components, then clicking them. The vast majority of interaction with this app is through button clicks since it's a mobile app.